File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.kt
class Solution {
    fun solution(n: Int): Long {
        var answer: Long = 0
        
        if (n == 1) {
            return 1%1234567
        } else if (n == 2) {
            return 2%1234567
        }
        
        var pre = 1L // 파이썬과 다르게 타입 명시를 해야한다...
        var next = 2L
        var temp = 0L
        
        for (i in 1..n-2) {
            temp = pre
            pre = next
            next = (temp + next)%1234567 // next에서 값을 계산 안해서 한참 헤맸다...
        }
        
        answer = next
        return answer
    }
}

// when을 사용한 결과값 반환...
class Solution {
    fun solution(n: Int): Long {
        var first: Long = 1
        var second: Long = 2
    return when(n) {
        1 -> first
        2 -> second
        else -> {
            var current: Long = first + second
            for (num in 3..n) {
                current = (first + second) % 1234567
                first = second
                second = current
            }
            current
        }
    }
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    return fibo(n)%1234567

def fibo(n):
    if n == 1:
        return 1
    elif n == 2:
        return 2
    else:
        pre, nex = 1, 2
        for i in range(n-2):
            pre, nex = nex, pre + nex
        return nex

# 한칸 또는 두칸밖에 못뜀
# 4라면 4를 2로 나눈수 2 == 2를 최대 몇번 사용하는지 알려줌
# 0번 -> 1개
# 1번 -> n-1개
# 2번 -> n-2개
# ...
# n/2 -> 1개
# 홀수이면 n/2 1개를 빼야함 
# -> 5부터 불가능

# 1 = 1
# 2 = 2
# 3 = 3
# 4 = 5
# 5 = 8
# 6 = 13
# -> 피보나치?
# Fn = Fn-1 + Fn-2